### PR TITLE
Finish Athena-compatible slice output

### PR DESCRIPTION
**Pull request by chengcli**
Replace the separate output_slicex1/output_slicex2/output_slicex3 flags with optional x1_slice/x2_slice/x3_slice coordinates. Coordinate presence now activates slicing, matching Athena's lower-inclusive, upper-exclusive slice ownership semantics. Validate slice coordinates against global mesh bounds and reject slice/sum conflicts for both YAML-loaded and programmatically constructed output options.

Implement OutputType slice and sum transforms for AthenaArray-backed output data. Apply transforms in x3, x2, x1 order, preserve output metadata, collapse transformed axes to one cell, track selected source-cell indices, and return false for blocks that do not intersect requested slice planes.

Integrate transforms into NetCDF output so non-intersecting blocks skip file creation while still participating in combine synchronization. Write selected cell-center coordinates, collapse sliced decomposition dimensions, and handle face-centered variables when their sliced axis becomes one cell.

Update Python bindings and stubs for Optional[float] slice coordinates, document slice YAML usage, and add tests for YAML activation, conflicts, boundary-cell selection, multi-axis slices, non-intersecting blocks, and NetCDF dimensions/coordinates.

Validated with:

- cmake --build build --target snap_release --parallel 8

- ctest --test-dir build --output-on-failure -R '^test_user_output\.release$'

- env CC=/opt/rh/gcc-toolset-9/root/usr/bin/gcc CXX=/opt/rh/gcc-toolset-9/root/usr/bin/g++ MAX_JOBS=8 /home/chengcli/pyenv/bin/python setup.py build_ext --inplace

The repository-wide build remains blocked by the existing NVCC host compiler being older than GCC 9.

### CODE REVIEW

File: tests/test_user_output.cpp [hunk 1]
```diff
@@ -2,8 +2,12 @@
 #include <gtest/gtest.h>
 
 // C/C++
+#include <netcdf.h>
+
```
[the remark this answers is not in the document]
Comment (chengcli) [reply tests/test_user_output.cpp:11]: Fixed in 7436d2d. The test now includes configure.h before conditionally including netcdf.h under NETCDFOUTPUT, matching the existing NetCDF-specific test guard.

File: src/output/output_type.cpp [hunk 2]
```diff
@@ -315,14 +321,174 @@ void OutputType::appendTensorOutput(std::string type, std::string name,
...
+    pnew->type = pdata->type;
+    pnew->name = pdata->name;
+    pnew->longname = pdata->longname;
+    pnew->units = pdata->units;
+
+    int nx4 = pdata->data.GetDim4();
+    int nx3 = pdata->data.GetDim3();
+    int nx2 = pdata->data.GetDim2();
+    int nx1 = pdata->data.GetDim1();
+    int source = *selected;
```
[the remark this answers is not in the document]
Comment (chengcli) [reply src/output/output_type.cpp:432]: Fixed in 7436d2d. Slice and sum transforms now resolve the metadata grid type for each node, map the requested physical axis onto the compact AthenaArray data dimension, and leave nodes unchanged when that physical axis is absent. Added regression coverage for --C profiles, -CC path slicing, and -CC path summation.
